packages/gestalt/src/Table.js

```diff
@@ -10,12 +10,18 @@ import TableRow from './TableRow.js';
 
 type Props = {|
   children: React.Node,
+  hasBorder?: boolean,
 |};
 
 export default function Table(props: Props) {
+  const { hasBorder, children } = props;
+
   return (
-    <Box overflow="scrollX">
-      <table className={styles.table}>{props.children}</table>
+    <Box
+      overflow="scrollX"
+      {...(hasBorder ? { borderSize: 'sm', rounding: 1 } : {})}
```

Comment:
I used a conditional here with spread props because the border should only show if hasBorder is true, but Box will always show a solid light gray border if anything (null/'none'/undefined) is passed in for borderSize.